plugins/org.eclipse.fordiac.ide.deployment.debug/src/org/eclipse/fordiac/ide/deployment/debug/DeploymentProcess.java

```diff
@@ -40,45 +35,51 @@
 import org.eclipse.fordiac.ide.deployment.exceptions.DeploymentException;
 import org.eclipse.fordiac.ide.model.libraryElement.INamedElement;
 
-public class DeploymentProcess extends PlatformObject implements IProcess, Callable<IStatus> {
+public class DeploymentProcess extends PlatformObject implements IProcess {
 
 	private final String name;
-	private final DownloadRunnable runnable;
 	private final ILaunch launch;
-	private final FutureTask<IStatus> task;
 	private final DeploymentStreamsProxy streamsProxy = new DeploymentStreamsProxy();
 	private final Map<String, String> attributes = new HashMap<>();
+	private final Job job;
 
 	public DeploymentProcess(final String name, final Set<INamedElement> selection, final ILaunch launch)
 			throws DeploymentException {
 		this.name = name;
 		this.launch = launch;
-		runnable = new DownloadRunnable(DeploymentCoordinator.createDeploymentdata(selection.toArray()), null,
-				streamsProxy, null);
-		task = new FutureTask<>(this);
+
+		final DownloadRunnable runnable = new DownloadRunnable(
+				DeploymentCoordinator.createDeploymentdata(selection.toArray()), null, streamsProxy, null);
+
+		job = new Job(name) {
+
+			@Override
+			protected IStatus run(final IProgressMonitor monitor) {
+				try {
+					runnable.run(monitor);
+					return Status.OK_STATUS;
+				} catch (final InterruptedException e) {
+					streamsProxy.getErrorStreamMonitor().message(Messages.DeploymentProcess_Terminated);
+					Thread.currentThread().interrupt();
+					return Status.error("Terminated"); //$NON-NLS-1$
+				} catch (final Exception t) {
+					streamsProxy.getErrorStreamMonitor()
+							.message(MessageFormat.format(Messages.DeploymentProcess_ExeceptionOccured, t.getMessage()));
+					return Status.error("Exception occurred", t); //$NON-NLS-1$
+				} finally {
+					fireTerminateEvent();
+				}
+			}
+
+		};
+		job.setUser(true);
+
```

Comment:
How about using `Job.create(String, IJobFunction)` and passing `DeploymentProcess::run` as a lambda instead of declaring an anonymous class in the constructor?

Reply:
I should have checked the api in more detail. Thx for pointing this out. Changed it.